### PR TITLE
Add &= and != to SpaceBeforeScript

### PR DESCRIPTION
**Pull request by igas**

### CODE REVIEW

File: lib/haml_lint/tree/script_node.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def parsed_script
       HamlLint::ParsedRuby.new(HamlLint::RubyParser.new.parse(script))
     end
 
-    # Returns the source for the script following the `-` marker.
+    # Returns the source for the script following the `=` marker.
```
Comment (michaelherold) [inline lib/haml_lint/tree/script_node.rb]: Good catch.

File: lib/haml_lint/linter/space_before_script.rb [hunk 2]
```diff
@@ -41,18 +41,23 @@ def visit_tag(node) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplex
     def visit_script(node)
       # Plain text nodes with interpolation are converted to script nodes, so we
       # need to ignore them here.
-      return unless document.source_lines[node.line - 1].lstrip.start_with?('=')
-      record_lint(node, MESSAGE_FORMAT % '=') if missing_space?(node)
+      line = document.source_lines[node.line - 1].lstrip
+      return unless line.start_with?('=', '&=', '!=')
```
Comment (michaelherold) [inline lib/haml_lint/linter/space_before_script.rb]: Huh. Today I learned that you can send multiple matchers to `start_with?`. 👍

**Comment by igas** [comment]
Yeah, there are few interesting differences, while in haml parser this 2 is "modifiers" to script, they behave differently, like changes in `missing_space?` for `= some_code` node text will include space, but with `&` / `!` in front of it - don't. ¯\_(ツ)_/¯

**Comment by trotzig** [comment]
I agree with @michaelherold in that this might be better as a separate linter.

**Comment by sds** [comment]
Closing due to lack of response. I agree this should be a separate linter for clarity's sake.

More than happy to merge once it's been extracted into a separate linter!

**Pull request closed without merging**